### PR TITLE
Change model logic. Disable cache. Implement fully issues loading

### PR DESCRIPTION
**Pull request by tormozz48**
@tadatuta @tavriaforever

### CODE REVIEW

File: server/model.js [hunk 1]
```diff
@@ -274,19 +231,17 @@ module.exports = {
...
 
-        var result = issues.filter(function(item) {
-            return item.number == options.number;
-        })[0];
+        if(issueNumber < 0) {
+            return vow.resolve(archive.getIssue(issueNumber));
+        }
 
-        return vow.resolve(result);
+        return github[getFnName(arguments.callee)].call(github, token, options);
```
Comment (tavriaforever) [inline server/model.js]: note for the future, it seems that not very good to use arguments.callee -http://stackoverflow.com/questions/103598/why-was-the-arguments-callee-caller-property-deprecated-in-javascript, you can use NFE http://learn.javascript.ru/named-function-expression#functions-nfe

File: server/model.js [hunk 2]
```diff
@@ -59,6 +55,12 @@ Archive.prototype = {
         return this.model.issues;
     },
 
+    getIssue: function(number) {
+        return this.getIssues().filter(function(item) {
+            return item.number == number;
```
Comment (tavriaforever) [inline server/model.js:62]:
```
===
```

File: server/model.js [hunk 3]
```diff
@@ -195,75 +154,73 @@ module.exports = {
...
+            //show only open issues and issues from archive
+            result = result.filter(function(item) {
+                return item.state !== 'closed';
+            });
 
-            result = result.filter(function(issueItem) {
-                var issueLabels = issueItem.labels.map(function(labelItem) {
-                    return labelItem.name || labelItem;
+            //filter by issue labels
+            if(options.labels && options.labels.length) {
```
Comment (tadatuta) [inline server/model.js]: `var labels = options.labels;`

File: server/model.js [hunk 4]
```diff
@@ -195,75 +154,73 @@ module.exports = {
...
+            }
 
-        //filter by updated date
-        if(options['since'] && _.isDate(options['since'])) {
-            result = result.filter(function(item) {
-                return (new Date(item['created_at'])).getTime() >= options['since'].getTime();
-            });
-        }
+            //filter by updated date
+            if(options['since'] && _.isDate(options['since'])) {
```
Comment (tadatuta) [inline server/model.js]: `var since = options.since;`

File: server/model.js [hunk 5]
```diff
@@ -195,75 +154,73 @@ module.exports = {
...
+                result = result.filter(function(item) {
+                    return (new Date(item['created_at'])).getTime() >= options['since'].getTime();
```
Comment (tadatuta) [inline server/model.js]: `item.created_at`

File: server/model.js [hunk 6]
```diff
@@ -195,75 +154,73 @@ module.exports = {
...
+                });
+            }
 
-        //sort results
-        sortField = (options.sort && /^(created|updated|comments)$/.test(options.sort))
+            //sort results
+            sortField = (options.sort && /^(created|updated|comments)$/.test(options.sort))
```
Comment (tadatuta) [inline server/model.js:239]: is it possible to get something else (not `created|updated|comments`) here?

File: server/model.js [hunk 7]
```diff
@@ -195,75 +154,73 @@ module.exports = {
...
-                bn = +b.number;
+            result = result.sort(function(a, b) {
+                var an = +a.number,
+                    bn = +b.number;
 
-            //separate gh and archive issues
-            if(an*bn < 0) {
-                return bn - an;
-            }
+                //separate gh and archive issues
```
Comment (tadatuta) [inline server/model.js:249]: why do we need to separate them?

File: server/model.js [hunk 8]
```diff
@@ -195,75 +154,73 @@ module.exports = {
...
-            if('comments' === sortField) {
-                return order*(+a[sortField] - +b[sortField]);
-            }else {
-                return order*((new Date(a[sortField + '_at'])).getTime() -
-                    (new Date(b[sortField + '_at'])).getTime());
-            }
-        });
+                if('comments' === sortField) {
+                    return order*(+a[sortField] - +b[sortField]);
+                }else {
```
Comment (tadatuta) [inline server/model.js]: possible to avoid `else` here

File: server/model.js [hunk 9]
```diff
@@ -195,75 +154,73 @@ module.exports = {
...
-        page = options.page || DEFAULT.page;
-        limit = options.per_page || DEFAULT.limit;
+            page = options.page || DEFAULT.page;
+            limit = options.per_page || DEFAULT.limit;
 
-        result = result.filter(function(item, index) {
-            return index >= limit*(page - 1) && index < limit*page
-        });
+            result = result.filter(function(item, index) {
+                return index >= limit*(page - 1) && index < limit*page
```
Comment (tadatuta) [inline server/model.js]: add spaces around `*`

File: server/model.js (hunk 1 above)
Comment (tadatuta) [inline server/model.js]: @tormozz48 please, explain this magic with `getFnName`

File: server/model.js (hunk 2 above)
Comment (tormozz48) [inline server/model.js:62]: No, here can be type transformation from string to number

File: server/model.js (hunk 6 above)
Comment (tormozz48) [inline server/model.js:239]: No, but it is good practice to verify incoming params

File: server/model.js (hunk 7 above)
Comment (tormozz48) [inline server/model.js:249]: We should show gh issues at first for all filters. After them we should show archive

Comment (tadatuta) [inline server/model.js:249]: there's no difference from user's point of view, so we don't need to separate them

**Pull request merged**